graphql_compiler/compiler/emit_sql.py

```diff
@@ -632,7 +632,7 @@ def _get_fold_outputs(self) -> List[Label]:
 
     def add_traversal(
         self,
-        join_descriptor: DirectJoinDescriptor,
+        join_descriptor: JoinDescriptor,
```

Comment:
I thought this seemed more reasonable than adding a type check in the test that was failing, but I'm not very familiar with the composite edge stuff so @bojanserafimov let me know if this doesn't actually make sense!

Reply:
This is not a user-friendly message (which edge uses composite keys?) so we should either make it user friendly, and remove the similar error check in `CompilationState.traverse`, or turn it into an assertion error.

Reply:
Alternatively, just add a comment at the other message explaining why its necessary even though the internal function does the same check: that the external check returns a friendlier message

Reply:
Ideally we'd have all our errors return a text span in the original query pointing at where the error is (so the UI can underline in red), but that's some polish work I don't think we'll get to